### PR TITLE
fix: compiler.webpack.ModuleFilenameHelpers type error

### PR DESCRIPTION
**Pull request by jerrykingxyz**
## Summary

Example:
Input:
``` typescript
import ModuleFilenameHelpers from "./lib/ModuleFilenameHelpers.js"

console.log(ModuleFilenameHelpers)
```
Ouput: (NodeNext)
``` typescript
const ModuleFilenameHelpers = require("./lib/ModuleFilenameHelpers.js")

// console undefined
console.log(ModuleFilenameHelpers.default)
```

Output: (CommonJS)
``` typescript
const ModuleFilenameHelpers = require("./lib/ModuleFilenameHelpers.js")

// console a object
console.log(ModuleFilenameHelpers)
```

## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Comment by jerrykingxyz** [comment]
!eco-ci

**Review by hardfist — DISMISSED** [review]

File: packages/rspack/tsconfig.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
     // TODO: Remove skipLibCheck when fix upstream type bug
     "skipLibCheck": true,
     "outDir": "dist",
-    "rootDir": "src"
+    "rootDir": "src",
+    "allowJs": true
```
Comment (xiaoxiangmoe) [inline packages/rspack/tsconfig.json:8]: `"allowJs": true` is extends from `../../tsconfig.base.json`

**Comment by jerrykingxyz** [comment]
!eco-ci

**Comment by hardfist** [comment]
it's indeed caused by nodeNext, let revert temporarily and dig into it later

**Review by hardfist — APPROVED** [review]

**Comment by hardfist** [comment]
let's merge it

**Comment by SoonIter** [comment]
@jerrykingxyz how can i test whether this bug is fixed in latest Typescript version `"5.5.3"`
https://github.com/web-infra-dev/rspack/pull/7211

**Comment by jerrykingxyz** [comment]
@SoonIter 
1. checkout this PR and replace `module` to NodeNext
2. try access compiler.webpack.ModuleFilenameHelper and it will throw a type error
3. upgrade the ts version and access compiler.webpack.ModuleFilenameHelper again

**Comment by SoonIter** [comment]
image

I think this issue is caused by `const ModuleFilenameHelpers = exports;`

this is a surprising writing of cjs code ... I think it has been fixed by renaming to `.ts` file

**Pull request merged**